### PR TITLE
[mergebot] set on green to true if it has ciflow

### PR DESCRIPTION
**Pull request by zengk95**
### Description

We've had reports of people adding ciflow labels and then merging with the expectation that it wouldn't merge until everything was passing. This wasn't how merge worked because it only works for mandatory jobs, so now based on that feedback, we're waiting for all jobs if there's a ciflow/* label on the PR. 

### Issue

n/a
### Testing

let tests run

Ran the trymerge locally and checked that on_green was true/false depending if the ciflow label was added

### CODE REVIEW

**Comment by zengk95** [comment]
Merge failed due to local variable 'on_green' referenced before assignment

**Comment by zengk95** [comment]
Merge failed due to Matched rule OSS CI, but PR #82110 has not been reviewed yet

**Comment by zengk95** [comment]
Merge failed due to Matched rule OSS CI, but PR #82110 has not been reviewed yet

**Comment by zengk95** [comment]
Merge failed due to Matched rule OSS CI, but PR #82110 has not been reviewed yet

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -364,6 +371,7 @@
     re.MULTILINE
 )
 RE_DIFF_REV = re.compile(r'^Differential Revision:.+?(D[0-9]+)', re.MULTILINE)
+CIFLOW_LABEL = re.compile(r"ciflow/*")
```
Comment (huydhn) [inline .github/scripts/trymerge.py:374]: Minor: This regex looks a bit off:

* If you just want it to match a string starting with `ciflow/`, r"^ciflow/" will do
* If you are looking to match the whole string like `ciflow/periodic`, the regex should be `ciflow/.*` or `ciflow/.+` or ciflow/\w*`

**Comment by ZainRizvi** [comment]
What's triggering these weird messages?

image

**Comment by kit1980** [comment]
> What's triggering these weird messages?

That's funny, the "successfully started a merge job" should be from the bot, the the PR's author.

File: .github/scripts/trymerge.py (hunk 1 above)
Comment (ZainRizvi) [reply .github/scripts/trymerge.py:374]: Huh, I'm surprised that the regex works the way it is right now (and it does work, I double checked).

Turns out that re.match() only matches from the start of the string.  Still, would be good to make the regex more explicit.

**Review by ZainRizvi — APPROVED** [review]
Looking good, other than the regex fix Huy suggested

**Comment by zengk95** [comment]
Those messages are coming from my local testing. Since it's using my GH token, it looks like i'm the one posting them.

**Comment by zengk95** [comment]
@huydhn currently -g and merge are essentially the same because mandatory checks == all checks on pull. We made merge only enforce mandatory only because
1. we were rolling this out, workflow by workflow and didn't want pull to be required yet at the beginning.
2. Maybe a second class of tests to come out that don't need to be required, but might provide some useful signal. These don't exist yet today.

**Review by malfet — APPROVED** [review]

**Review by huydhn — APPROVED** [review]

**Pull request closed without merging**